File: src/main/org/nlogo/api/Let.java
package org.nlogo.api ;

import java.util.Collections;
import java.util.List;

public final strictfp class Let
{
	public final String varName ;
	public final int startPos ;
	public final int endPos ;
	public final List<Let> children ;
	public Let()
	{
		varName = null ;
		startPos = -1 ;
		endPos = -1 ;
		children = Collections.emptyList() ;
	}
	public Let( String varName , int startPos , int endPos , List<Let> children )
	{
		this.varName = varName ;
		this.startPos = startPos ;
		this.endPos = endPos ;
		this.children = children ;
	}
}

File: netlogo-gui/src/main/api/Let.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.api

import java.util.{ Collections, List => JList }

final case class Let(varName: String, start: Int, end: Int, children: JList[Let]) {
  def this() = this(null, -1, -1, Collections.emptyList[Let])
  def this(varName: String, start: Int, end: Int) = this(varName, start, end, Collections.emptyList[Let])
}
